File: lib/midi/Synthesizer.ts
import { MidiValue } from './MidiValue';
import { PlayNoteParameters } from './PlayNoteParameters';

interface Synthesizer {
  setController: ({ controller, value }: {
    controller: MidiValue;
    value: MidiValue;
  }) => void;

  selectSound: ({ value }: {
    value: MidiValue;
  }) => void;

  playNote: ({ noteValue, velocity }: PlayNoteParameters) => void;

  stop: () => void;
}

export { Synthesizer };

File: synthesizer/synthesizer.go
package synthesizer

import "gitlab.com/gomidi/midi/v2"

type Synthesizer interface {
	SelectSound(bank, sound uint8)
	SetController(controller, value uint8)
	PlayNote(note midi.Note, velocity uint8)
	StopNote(note midi.Note)
	StopAllNotes()
}
